android/core/src/main/java/com/pega/vehicledamagemodeling/VehicleDamageModeling.java

```diff
@@ -39,50 +41,65 @@
 import static com.badlogic.gdx.graphics.g3d.attributes.ColorAttribute.AmbientLight;
 
 public class VehicleDamageModeling extends ApplicationAdapter {
-    private PerspectiveCamera perspectiveCamera;
-    private LimitedCameraInputController cameraController;
     private ModelBatch modelBatch;
+    private Environment environment;
+    private PerspectiveCamera camera;
+    private LimitedCameraInputController cameraController;
+    private SelectionService selectionService = new SelectionService(new SelectedPartsRepository(), new Parser());
     private AssetManager assets;
     private Array<ModelInstance> instances = new Array<>();
-    private Environment environment;
-    private boolean loading;
-    private SelectionService selections = new SelectionService(new SelectedPartsRepository(), new Parser());
     private final VehicleDamageReportCallback callback;
+    private JsonObject jsonWithSelectedParts;
+
+    private boolean loading;
+
     private static final String MODEL_FILE_NAME = "model.2.1.obj";
 
     public VehicleDamageModeling(VehicleDamageReportCallback callback) {
         this.callback = callback;
     }
 
-    public VehicleDamageModeling(JsonObject report, VehicleDamageReportCallback callback) {
+    public VehicleDamageModeling(JsonObject jsonWithSelectedParts, VehicleDamageReportCallback callback) {
         this(callback);
-        selections.attachJson(report);
+        this.jsonWithSelectedParts = jsonWithSelectedParts;
     }
 
     @Override
     public void create () {
         modelBatch = new ModelBatch();
+
```

Comment:
Dla mnie tej pustej linii moze nie byc. Ale jesli wedlug Ciebie zwieksza czytelnosc to spoko